st2api/st2api/controllers/v1/auth.py

```diff
@@ -97,11 +101,27 @@ def get_all(self, requester_user, show_secrets=None):
         """
         mask_secrets = self._get_mask_secrets(show_secrets=show_secrets,
                                               requester_user=requester_user)
-        api_key_dbs = ApiKey.get_all()
-        api_keys = [ApiKeyAPI.from_model(api_key_db, mask_secrets=mask_secrets)
-                    for api_key_db in api_key_dbs]
 
-        return api_keys
+        if limit and int(limit) > self.max_limit:
```

Comment:
Yes, we are using this here in the resource controller: https://github.com/StackStorm/st2/blob/1043ed8b2659625bb02b58e52639d7832a7a8974/st2api/st2api/controllers/resource.py#L160, but are not using it as a base class in this case.